### PR TITLE
Handle obsolete InterceptorsPreview with .NET 8 RC2 SDK

### PR DESCRIPTION
**Pull request by mthalman**
When building the runtime repo using the latest build of .NET 8 RC2 SDK, it produces the following error:

`
/vmr/src/runtime/artifacts/source-build/self/src/src/libraries/Microsoft.Extensions.Logging.Configuration/src/Microsoft.Extensions.Configuration.Binder.SourceGeneration/Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator/BindingExtensions.g.cs(33,10): error CS9137: The 'interceptors' experimental feature is not enabled in this namespace. Add '<InterceptorsPreviewNamespaces>$(InterceptorsPreviewNamespaces);Microsoft.Extensions.Configuration.Binder.SourceGeneration</InterceptorsPreviewNamespaces>' to your project. [/vmr/src/runtime/artifacts/source-build/self/src/src/libraries/Microsoft.Extensions.Logging.Configuration/src/Microsoft.Extensions.Logging.Configuration.csproj::TargetFramework=net8.0]
`

This is due to a new change from Roslyn: https://github.com/dotnet/roslyn/pull/69848. It replaces the usage of the `InterceptorsPreview` feature with an `InterceptorsPreviewNamespaces` property that needs to specify the applicable namespace.

The use of `InterceptorsPreview` in the project is still needed in order to work with the RC1 SDK. Both need to be specified as support for both versions of the SDK is required to support source-build scenarios. A separate issue will be logged as follow-up to remove the obsolete `InterceptorsPreview` feature usage.

This change is needed for RC2 release.

### CODE REVIEW

**Comment by tarekgh** [comment]
@mthalman isn't the SDK already enable `InterceptorsPreviewNamespaces` when specifying `EnableConfigurationBindingGenerator` in the project? 
https://github.com/dotnet/sdk/blob/96b9f735229a3b136c15bf8b47d5e9c0ca2cc1a4/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.targets#L66

Or does the used SDK build not have this change? 

CC @layomia @ericstj

**Review by carlossanlop — APPROVED** [review]
@mthalman thanks for sending this change. I had also fixed it in the dotnet/roslyn dependency flow PR, but it's slightly differently: https://github.com/dotnet/runtime/pull/92150/commits/8013838eba1c94f84182b023252babd6f6386d8a

Unfortunately, another CI failure showed up, so I suppose the value I selected for InterceptorsPreviewNamespaces was not the right one? https://github.com/dotnet/runtime/pull/92150#issuecomment-1724372265

Anyway, if the CI is green in this PR, we can merge it before the roslyn deps flow and I can revert my change in that PR.

Tentatively approving.

cc @ericstj @layomia @tarekgh

**Comment by carlossanlop** [comment]
Also - Marking this as `servicing-approved` as it is an infra change (tell-mode).

File: src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj [hunk 1]
```diff
@@ -4,7 +4,9 @@
     <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
     <!-- Type not supported; property on type not suppported -->
     <NoWarn>SYSLIB1100,SYSLIB1101</NoWarn>
+    <!-- TODO: Remove InterceptorsPreview feature after 8.0 RC2 SDK is used for build -->
     <Features>$(Features);InterceptorsPreview</Features>
+    <InterceptorsPreviewNamespaces>$(InterceptorsPreviewNamespaces);Microsoft.AspNetCore.Http.Generated</InterceptorsPreviewNamespaces>
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj:9]: I don't think this is the right namespace for this repo.  Shouldn't we use the Configuration namespace, like Carlos's PR?

**Comment by ericstj** [comment]
> The use of InterceptorsPreview in the project is still needed in order to work with the RC1 SDK

Ahh - source build is a good case.  @carlossanlop this was what I was guessing at in our chat earlier when trying to think of something that might need to use the old SDK on the latest codebase.  Source build bootstrapping.

**Comment by carlossanlop** [comment]
I might be able to address this problem in the the two dotnet/roslyn deps flow PRs (main and 8.0), thanks to @ericstj help. If the CI looks green, I'll merge those ones, and we can close this (I copied the TODO comment there).

File: src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj (hunk 1 above)
Comment (mthalman) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj:9]: Whoops. Copy-paste error after making similar changes in aspnetcore.

**Comment by carlossanlop** [comment]
The deps flow PR has the issue handled, so I think it's safe to close this PR, @mthalman : https://github.com/dotnet/runtime/pull/92149

Unfortunately, fixing it uncovered a test failure in Microsoft.Extensions.Configuration.Binder. We should take care of it in the deps flow PR too.

**Pull request closed without merging**